packages/shared-components/src/components/onto-repository-selector/onto-repository-selector.tsx

```diff
@@ -140,4 +153,86 @@ export class OntoRepositorySelector {
     }
     return ``;
   }
+
+  private getRepositoryTooltipFunction(repository: Repository): () => Promise<string> {
+    return () => {
+      if (!repository) {
+        return Promise.resolve('');
+      }
+      return this.repositoryService.getRepositorySizeInfo(repository)
+        .then((repositorySizeInfo: RepositorySizeInfo) => this.buildRepositoryTooltipHtml(repository, repositorySizeInfo));
+    }
+  }
+
+  private buildRepositoryTooltipHtml(repository: Repository, repositorySizeInfo: RepositorySizeInfo): string {
+    let repositoryTooltipHtml = `
```

Comment:
I wonder if this is the best way to compose these templates here and the other one below. I also think that these could eventually be a standalone component itself